### PR TITLE
Adding implementation of 'az quantum workspace create' command.

### PR DESCRIPTION
**Pull request by ricardo-espinoza**
Adding initial implementation of 'az quantum workspace create' command.

_________________
Command
    az quantum workspace create : Creates a new Azure Quantum workspace.
        Command group 'quantum' is in preview. It may be changed/removed in a future
        release.
Arguments
    --location -l         : Location. Values from: `az account list-locations`. You can configure
                            the default location using `az configure --defaults
                            location=<location>`.
    --resource-group -g   : Name of resource group. You can configure the default group using `az
                            configure --defaults group=<name>`.
    --storage_account -sa : Name of the storage account to be used by a quantum workspace.
    --workspace-name -w   : Name of the Quantum Workspace. You can configure the default workspace
                            using `az quantum workspace set`.

### CODE REVIEW

File: src/quantum/azext_quantum/_help.py [hunk 1]
```diff
@@ -54,6 +54,9 @@
       - name: Get the list of Azure Quantum workspaces available
         text: |-
             az quantum workspace list
+      - name: Create a new Azure Quantum workspace
+        text: |-
+            az quantum workspace create -g MyResourceGroup -w MyWorkspace -l MyLocation -sa MyStorageAccountName
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_help.py:59]: Ah, thanks, this is good to know for the other QDK work: the CLI refers to this as "location", not "region". I'll be consistent with that in the other APIs as well.

File: src/quantum/azext_quantum/_params.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 
 def load_arguments(self, _):
     workspace_name_type = CLIArgumentType(options_list=['--workspace-name', '-w'], help='Name of the Quantum Workspace. You can configure the default workspace using `az quantum workspace set`.', id_part=None, required=False)
+    storage_account_name_type = CLIArgumentType(options_list=['--storage_account', '-sa'], help='Name of the storage account to be used by a quantum workspace.')
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_params.py:12]: Am I understanding correctly that things like `-g` (for resource group) and `-l` (for location) are handled by some common Azure CLI code? So the only things that need to be listed here are things specific to `az quantum`?

File: src/quantum/azext_quantum/_params.py [hunk 3]
```diff
@@ -19,6 +20,7 @@ def load_arguments(self, _):
 
     with self.argument_context('quantum workspace') as c:
         c.argument('workspace_name', workspace_name_type)
+        c.argument('storage_account', storage_account_name_type)
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/_params.py:23]: It is sort of unfortunate that there is another parameter called `storage` which means something totally different. Not sure there's much that can be done though.

File: src/quantum/azext_quantum/operations/workspace.py [hunk 4]
```diff
@@ -43,6 +46,39 @@ def save(self, cmd):
...
+    # Replace this with user specified providers as part of task:
+    # https://ms-quantum.visualstudio.com/Quantum%20Program/_workitems/edit/16184
+    prov = Provider()
+    prov.provider_id = "Microsoft"
+    prov.provider_sku = "Basic"
+    qw.providers = [prov]
+    # Allow the system to assign the workspace identity
+    qw.identity = QuantumWorkspaceIdentity()
+    qw.identity.type = "SystemAssigned"
+    qw.location=location
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/operations/workspace.py]:
```suggestion
    qw.location = location
```

File: src/quantum/azext_quantum/operations/workspace.py [hunk 5]
```diff
@@ -43,6 +46,39 @@ def save(self, cmd):
...
+    qw.storage_account = "/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Storage/storageAccounts/{2}".format(info.subscription, info.resource_group, storage_account)
```
Comment (rmshaffer) [inline src/quantum/azext_quantum/operations/workspace.py]: nit, for readability (ignore if this code has to support Python versions <3.6, when `f`-strings were introduced... 3.5 is out-of-support, but I'm not sure whether Azure CLI guarantees a particular Python version?)
```suggestion
    qw.storage_account = f"/subscriptions/{info.subscription}/resourceGroups/{info.resource_group}/providers/Microsoft.Storage/storageAccounts/{storage_account}"
```

**Review by rmshaffer — APPROVED** [review]

File: src/quantum/azext_quantum/_params.py (hunk 2 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/_params.py:12]: Right. Some arguments are inherited from the core `az cli` code.

File: src/quantum/azext_quantum/_params.py (hunk 3 above)
Comment (ricardo-espinoza) [reply src/quantum/azext_quantum/_params.py:23]: Yes, I consider reusing storage for this, but this seemed dangerous, since it has its own purpose.

**Pull request merged**